custom/nic_compliance/forms.py

```diff
@@ -0,0 +1,37 @@
+from django import forms
+from django.utils.translation import ugettext as _
+from django.core.exceptions import ValidationError
+from django.contrib.auth import password_validation
+
+
+class EncodedPasswordChangeForm(object):
+    def clean_new_password1(self):
+        from corehq.apps.domain.forms import clean_password
+        from corehq.apps.hqwebapp.utils import decode_password
+        new_password = decode_password(self.cleaned_data.get('new_password1'))
+        if new_password == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='new_password1_empty',
+            )
+
+        return clean_password(new_password)
```

Comment:
I think this should be under an if statement: ` if self.project.strong_mobile_passwords:`